src/p4pktgen/switch/runtime_CLI.py

```diff
@@ -827,77 +969,68 @@ def exactly_n_args(self, args, n):
                 "Wrong number of args, expected %d but got %d" % (n, len(args))
             )
 
+    def parse_int(self, arg, name):
+        try:
+            return int(arg)
+        except:
+            raise UIn_Error("Bad format for {}, expected integer".format(name))
+
     def _complete_res(self, array, text):
         res = sorted(array.keys())
         if not text:
             return res
         return [r for r in res if r.startswith(text)]
 
-    # CHANGE_FROM_ORIGINAL: The @handle_bad_input decorator that was
-    # just before 'def do_show_tables' modifies the behavior of
-    # do_show_tables (and any other method it decorates) by handling
-    # various kinds of exceptions it can raise, printing a message,
-    # and preventing the exception from being raised to the caller.
-
-    # This makes sense in the context of calling these methods from
-    # the simple_switch_CLI program, where clearer error messages, and
-    # avoiding simple_switch_CLI from exiting due to a raised
-    # exception, are desirable.
-
-    # In the context of calling these methods from a set of automated
-    # tests, it is far more useful for the exception to be 'passed on'
-    # to the caller, because it makes mistakes in calling these
-    # methods far more obvious.
-    #@handle_bad_input
+    @handle_bad_input
```

Comment:
Have you found the @handle_bad_input not to make it inconvenient for seeing exceptions from calls that return errors?  Or perhaps that was growing pains early in the discovery process of us learning the correct way to make such calls, and such errors are now rare.

Reply:
See comment on L758.  It's a little hard to see when skimming as it's just and extra sentence added onto the comment block that was removed from this location.

There is a different issue now.  The exceptions raised by this module are not hashable and so in python3 pytest errors out if it has to handle them (as in, it exits, subsequent tests do not get run).
The handle_bad_input decorator is now responsible for extracting the message and putting it into a standard exception.

Reply:
Makes sense.  Sounds fine to me, then.